package/kernel/mac80211/patches/ath11k/200-wifi-ath11k-add-support-for-QAM-256-in-2.4GHz-802.11.patch

```diff
@@ -0,0 +1,57 @@
+From 462b0e7ce7cadef5c14bb16134e43573b169c9b5 Mon Sep 17 00:00:00 2001
+From: Christian Marangi <ansuelsmth@gmail.com>
+Date: Wed, 14 Jun 2023 09:44:29 +0200
+Subject: [PATCH 4/4] wifi: ath11k: add support for QAM-256 in 2.4GHz 802.11n
+
+Ath11k supports non-standard QAM-256 in 2.4GHz in 802.11n. Permit for
+supported client to benefits from these extra bandwidth by enabling
+support for it.
+
+Signed-off-by: Christian Marangi <ansuelsmth@gmail.com>
+---
+ drivers/net/wireless/ath/ath11k/mac.c | 7 +++++--
+ drivers/net/wireless/ath/ath11k/wmi.c | 2 ++
+ 2 files changed, 7 insertions(+), 2 deletions(-)
+
+diff --git a/drivers/net/wireless/ath/ath11k/mac.c b/drivers/net/wireless/ath/ath11k/mac.c
+index c947d1c8d8c1..1cadfdd50afc 100644
+--- a/drivers/net/wireless/ath/ath11k/mac.c
++++ b/drivers/net/wireless/ath/ath11k/mac.c
+@@ -2773,9 +2773,9 @@ static void ath11k_peer_assoc_h_phymode(struct ath11k *ar,
+ 		} else if (sta->deflink.vht_cap.vht_supported &&
+ 			   !ath11k_peer_assoc_h_vht_masked(vht_mcs_mask)) {
+ 			if (sta->deflink.bandwidth == IEEE80211_STA_RX_BW_40)
+-				phymode = MODE_11AC_VHT40;
++				phymode = MODE_11AC_VHT40_2G;
+ 			else
+-				phymode = MODE_11AC_VHT20;
++				phymode = MODE_11AC_VHT20_2G;
```

Comment:
These are constants from `ath10k/wmi.h`, right?
What would be the equivalent constants for `mt76` (and other WiFi ICs)? is it `enum mt76_phy_type` from `mt76.h`?

Reply:
it's driver specific and really how the fw implemented them... can totally be not necessary... in fact ath10k doesn't used them originally... so it's just for consistency.

Reply:
It's not needed for mt76, firmware doesn't care. You could reuse the 5g code and it will work OK